### ISSUE TITLE
`Suggest questions...` button spins when clicked in the Response rate limit modal

### ISSUE THREAD

**Issue by MadhaviSeelam** [brave/brave-browser#36138]
## Description 

Found while testing https://github.com/brave/brave-browser/issues/35503.  When clicked on `Suggest questions...` button in the Response rate limit modal, it spins.  Perhaps the button should be disabled/hidden?

## Steps to Reproduce

1. Install 1.63.155
2. launch Brave (without commandline flags)
3. Open Leo from sidebar
4. Load wikipedia.com and summarize the page
5.  Query Leo until you hit the Response rate limit modal
6.  Switched to `Llama` and queried
7.  Switched again to Claude Instant and queried until Response rate limit modal returned
8.  clicked `Suggest questions..`  button

## Actual result:

`Suggest questions...` button spins

https://github.com/brave/brave-browser/assets/98358127/87864e37-d666-4ff4-98ca-3b6b83873fcc

## Expected result:
  Perhaps the button should be disabled/hidden?

## Reproduces how often: 

Easily

## Brave version (brave://version info)

```
Brave | 1.63.155 Chromium: 122.0.6261.29 (Official Build) (64-bit)
-- | --
Revision | b2be2c6f5b7340629f672ce706f5bfd62b113f82
OS | Windows 11 Version 23H2 (Build 22631.3155)
```

## Version/Channel Information:

- Can you reproduce this issue with the current release? N/A
- Can you reproduce this issue with the beta channel? Yes
- Can you reproduce this issue with the nightly channel? N/A

## Other Additional Information:

- Does the issue resolve itself when disabling Brave Shields? 
- Does the issue resolve itself when disabling Brave Rewards? 
- Is the issue reproducible on the latest version of Chrome? 

## Miscellaneous Information:

 @mattmcalister @LorenzoMinto 
cc: @rebron @stephendonner

**Comment by nullhook** [comment]
this issue should be fixed with: https://github.com/brave/brave-core/pull/22269

can you confirm @MadhaviSeelam ?

**Comment by MadhaviSeelam** [comment]
> this issue should be fixed with: [brave/brave-core#22269](https://github.com/brave/brave-core/pull/22269)
> 
> can you confirm @MadhaviSeelam ?

@nullhook - verified in the below version.  Seem to be it's still spinning or it spins a bit.  Inconsistent behavior. 
```
Brave | 1.65.53 Chromium: 122.0.6261.94 (Official Build) nightly (64-bit)
-- | --
Revision | 4d7a1aecb1f9b3194d1b89cac2329c33eaa15a58
OS | Windows 11 Version 23H2 (Build 22631.3155)
```
https://github.com/brave/brave-browser/assets/98358127/2b0d0443-573b-4aec-b482-1e94ab09e632

**Comment by nullhook** [comment]
> it spins a bit

i should've clarified when i stated 'should be fixed.' there's a subtle difference before the PR merged and after. prior to the PR, the button was stuck in a loading state, and after the PR, we silently reset the button's state, so the user can continue further. we will have a fully designed UI state for this case. here's the issue: https://github.com/brave/brave-browser/issues/36540

so, as a quick fix, did this resolve the issue? let me know if you'd like to close this.

**Comment by MadhaviSeelam** [comment]
> > it spins a bit
> 
> i should've clarified when i stated 'should be fixed.' there's a subtle difference before the PR merged and after. prior to the PR, the button was stuck in a loading state, and after the PR, we silently reset the button's state, so the user can continue further. we will have a fully designed UI state for this case. here's the issue: #36540
> 
> so, as a quick fix, did this resolve the issue? let me know if you'd like to close this.

Looks good now on a new profile.   Earlier I did see it stuck in loading state but not seeing any more.

https://github.com/brave/brave-browser/assets/98358127/65ab9c0e-f738-4486-9597-fb5fd5037af2

**Comment by nullhook** [comment]
OK, closing this because the 'suggest questions' button no longer spins, as this was the original issue reported. A more comprehensive solution will be addressed at #36540
